library/src/com/google/maps/android/kml/KmlContainer.java

```diff
@@ -0,0 +1,254 @@
+package com.google.maps.android.kml;
+
+import com.google.android.gms.maps.model.GroundOverlay;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Represents a KML Document or Folder.
+ */
+public class KmlContainer {
+
+    private final HashMap<String, String> mProperties;
+
+    private final HashMap<KmlPlacemark, Object> mPlacemarks;
+
+    private final ArrayList<KmlContainer> mContainers;
+
+    private final HashMap<KmlGroundOverlay, GroundOverlay> mGroundOverlays;
+
+    private final HashMap<String, String> mStyleMap;
+
+    private HashMap<String, KmlStyle> mStyles;
+
+    private String mContainerId;
+
+    public KmlContainer() {
+        mProperties = new HashMap<String, String>();
+        mPlacemarks = new HashMap<KmlPlacemark, Object>();
+        mStyles = new HashMap<String, KmlStyle>();
+        mStyleMap = new HashMap<String, String>();
+        mContainers = new ArrayList<KmlContainer>();
+        mGroundOverlays = new HashMap<KmlGroundOverlay, GroundOverlay>();
+        mContainerId = null;
+    }
+
+    /**
+     * Gets the Container ID if it is specified
+     *
+     * @return Container ID or null if not set
+     */
+    public String getContainerId() {
+        return mContainerId;
+    }
+
+    /**
+     * Assigns a string value to the Contianer
+     *
+     * @param containerId   String value to assign to the contianer
+     */
+    public void setContainerId(String containerId) {
```

Comment:
I've had a thought about this and it's something I want to change later. The reason why we've currently got set methods is because XML is parsed sequentially.